### DISCUSSION TITLE
'PMcycle' and 'PMfinal' object not saved

### DISCUSSION THREAD

**Discussion by christyny** [LAPKB/Pmetrics#64] [Q&A] [answered]
Hello, I'm a graduate student from China and I've recently encountered some problems with Pmetrics(version 1.9.7).When I run, I get the following errors in the terminal:

WARNING: error in extraction of cycle information; 'PMcycle' object not saved.
WARNING: error in extraction of final cycle parameter values; 'PMfinal' object not saved.

If I try to make the objects manually, I get the following errors:

 npag = NPparse(outfile = "NP_RF0001.TXT")
Parsing NPAG results...
Error in NPparse(outfile = "NP_RF0001.TXT") : NP_RF0001.TXT not found.

I I found this file NP_RF0001.TXT in the folder“run /1/outputs”,but it dosen't work.

**Comment by mhovd** [comment]
Potentially related to [this issue ](https://github.com/LAPKB/Pmetrics/issues/58#issue-1225718825) ?

**Comment by Siel** [comment]
It seems so, @christyny which operating system are you using?

**Reply by christyny** [reply]
I use the operating system Windows 10.

**Reply by christyny** [reply]
When I run the plot(final.1) and plot(cycle.1),the following errors occur.

> plot(final.1)
Error in plot.window(...) : 'xlim' value cannot be infinite
In addition: Warning messages:
1: In min(x) : no non-missing arguments to min; returning Inf
2: In max(x) : no non-missing arguments to max; returning -Inf
3: In min(x) : no non-missing arguments to min; returning Inf
4: In max(x) : no non-missing arguments to max; returning -Inf

> plot(cycle.1,cex.leg=0.8)
Error in plot.window(...) : 'xlim' value cannot be infinite
In addition: Warning messages:
1: In min(x) : no non-missing arguments to min; returning Inf
2: In max(x) : no non-missing arguments to max; returning -Inf
3: In min(x) : no non-missing arguments to min; returning Inf
4: In max(x) : no non-missing arguments to max; returning -Inf
5: In plot.window(...) : "cex.leg" is not a figure parameter

**Comment by Siel** [comment, marked as answer]
I'm still unable to reproduce the error, could you share with me the dataset/model you are having issues with?

**Reply by christyny** [reply]
I reinstalled R 4.1.3 and the problem is solved, thanks.

**Reply by mnneely** [reply]
Great thanks for letting us know!

**Comment by Siel** [comment]
This bug was fixed on the Version2 branch that is going to be released in the near future

**Reply by yoannc40** [reply]
I just installed Pmetrics 1.9.7 and I have the exact same problem as @christyny. My configuration is windows 10 and R version 4.2.1. I tried to install it on another windows 11 computer and I have exactly the same problem. All this with the "examples" file. Does anyone have a solution? What is version 2? Pmetrics?

Thanks for your answers
![capture 1](https://user-images.githubusercontent.com/8640718/188517326-3e05f485-5b7b-4524-bd04-742feb35c740.png)
![image](https://user-images.githubusercontent.com/8640718/188517385-a86fe2cf-6b32-409d-a315-18c26bbc83ab.png)
consequently, tabs on html convergence plots, marginal plots and paramter values are empty + no PDF...

**Reply by Siel** [reply]
Hi @yoannc40, there are two solutions at the time:

- Install Pmetrics 2.0 using `devtools::install_github("LAPKB/Pmetrics@Version2")`, or
- Downgrade R to 4.1.3 to keep using Pmetrics 1.9.7
Hope it helps!

**Reply by yoannc40** [reply]
Great, it works!!!

Thx+++
